client/src/js/services/Session.js

```diff
@@ -40,11 +40,12 @@ function SessionService($sessionStorage, $http, $location, util, $rootScope) {
 
   // set the user, enterprise, and project for the session
   // this should happen right after login
-  function create(user, enterprise, project, path) {
+  function create(user, enterprise, project, path, lang) {
     $storage.user = user;
     $storage.enterprise = enterprise;
     $storage.project = project;
     $storage.path = path;
+    $storage.lang = lang;
```

Comment:
I don't think you need to add these language values to the session of it is being send in the query string for now - this could be removed.